PackageViewModel/Utilities/FileHelper.cs

```diff
@@ -169,5 +171,157 @@ public static bool IsAssembly(string path)
                    path.EndsWith(".winmd", StringComparison.OrdinalIgnoreCase) ||
                    path.EndsWith(".exe", StringComparison.OrdinalIgnoreCase);
         }
+
+
+        [Flags]
+        private enum SHGFI
+        {
+            /// <summary>
+            /// Get icon
+            /// </summary>
+            Icon = 0x000000100,
+
+            /// <summary>
+            /// get display name
+            /// </summary>
+            DisplayName = 0x000000200,
+
+            /// <summary>
+            /// Get type name
+            /// </summary>
+            TypeName = 0x000000400,
+
+            /// <summary>
+            /// Get attributes
+            /// </summary>
+            Attributes = 0x000000800,
+
+            /// <summary>
+            /// Get icon location
+            /// </summary>
+            IconLocatin = 0x000001000,
+
+            /// <summary>
+            /// Return exe type
+            /// </summary>
+            ExeType = 0x000002000,
+
+            /// <summary>
+            /// Get system icon index
+            /// </summary>
+            SysIconIndex = 0x000004000,
+
+            /// <summary>
+            /// Put a link overlay on icon
+            /// </summary>
+            LinkOverlay = 0x000008000,
+
+            /// <summary>
+            /// Show icon in selected state
+            /// </summary>
+            Selected = 0x000010000,
+
+            /// <summary>
+            /// Get only specified attributes
+            /// </summary>
+            Attr_Specified = 0x000020000,
+
+            /// <summary>
+            /// Get large icon
+            /// </summary>
+            LargeIcon = 0x000000000,
+
+            /// <summary>
+            /// Get small icon
+            /// </summary>
+            SmallIcon = 0x000000001,
+
+            /// <summary>
+            /// Get open icon
+            /// </summary>
+            OpenIcon = 0x000000002,
+
+            /// <summary>
+            /// Get shell size icon
+            /// </summary>
+            ShellIconize = 0x000000004,
+
+            /// <summary>
+            /// pszPath is a pidl
+            /// </summary>
+            PIDL = 0x000000008,
+
+            /// <summary>
+            /// Use passed dwFileAttribute
+            /// </summary>
+            UseFileAttributes = 0x000000010,
+
+            /// <summary>
+            /// Apply the appropriate overlays
+            /// </summary>
+            AddOverlays = 0x000000020,
+
+            /// <summary>
+            /// Get the index of the overlay in the upper 8 bits of the iIcon
+            /// </summary>
+            OverlayIndex = 0x000000040,
+
+            /// <summary>
+            /// The handle that identifies a directory.
+            /// </summary>
+            FILE_ATTRIBUTE_DIRECTORY = 0x10,
+        }
+
+        /// <summary>
+        /// Maximal Length of unmanaged Windows-Path-strings
+        /// </summary>
+        private const int MAX_PATH = 260;
```

Comment:
This isn't really true anymore, FYI.

Reply:
Thanks, so the ```SizeConst = MAX_PATH``` should be removed from attribute?

Reply:
I'm not completely sure what the right answer is, I'll look into it.

Reply:
The Win32 API sure does look like it is documented to be limited to `MAX_PATH`. https://docs.microsoft.com/en-us/windows/desktop/api/shellapi/ns-shellapi-_shfileinfow

Not sure how this API was updated to account for new limit, but you can't just remove it or change it because it will affect the size of the struct.